### ISSUE TITLE
pi05 training is not converged?

### ISSUE THREAD

**Issue by xianglunkai** [mit-han-lab/vlash#16]
[Report _ vlash – Weights & Biases.pdf](https://github.com/user-attachments/files/24445007/Report._.vlash.Weights.Biases.pdf)

# VLASH Training Configuration for PI05 with Async Inference

# Policy configuration
policy:
  type: pi05
  pretrained_path: /workspace/lerobot/pretrain_model/pi05_base
  push_to_hub: false
  dtype: bfloat16
  device: cuda:0
  state_cond: true
  input_features: {
        "observation.images.high": {"type": "VISUAL", "shape": [480, 640, 3]},
        "observation.images.left": {"type": "VISUAL", "shape": [480, 640, 3]},
        "observation.images.right": {"type": "VISUAL", "shape": [480, 640, 3]},
        "observation.state": {"type": "STATE", "shape": [14]}}
  output_features: {"action": {"type": "ACTION", "shape": [14]}}
  chunk_size: 30
  n_action_steps: 30
  compile_model: true
  gradient_checkpointing: true

# Dataset configuration
dataset:
  repo_id: /workspace/huggingface/lerobot/handover_bottle_v30
  # video_backend: torchcodec

# Training parameters
output_dir: outputs/train/pi05_async
job_name: pi05_async
batch_size: 8
steps: 50000
num_workers: 4
seed: 1000
rename_map: {"observation.images.cam_high":"observation.images.high","observation.images.cam_left_wrist":"observation.images.left","observation.images.cam_right_wrist":"observation.images.right","observation.state":"observation.state","action":"action"}

# # Optimizer configuration
# use_policy_training_preset: false
# optimizer:
#   type: adamw
#   lr: 2.5e-5
#   betas: [0.9, 0.95]
#   weight_decay: 1.0e-10

# # Learning rate scheduler
# scheduler:
#   type: cosine_decay_with_warmup
#   num_warmup_steps: 1000
#   peak_lr: 5.0e-5
#   decay_lr: 2.5e-6
#   num_decay_steps: 50000

# Checkpointing
save_checkpoint: true
save_freq: 10000

# Logging
log_freq: 100
wandb:
  enable: true
  project: vlash
  entity: null
  disable_artifact: true

# Maximum temporal delay steps
# Set to N to enable random delays from [0, N]
max_delay_steps: 8

# Enable shared observation optimization
# When true, trains all offsets [0, max_delay_steps] together with shared
# observation embeddings instead of randomly sampling one offset per sample
